File: demos/js/src/demo_random-walks.ts
// visualize `data`
const data = new Array<number>();
let curValue = 0;

for (let j = 0; j < 100; j++) {
	addManyRandomValues();
}

function addManyRandomValues() {
	for (let i = 0; i < 100; i++) {
		addRandomValue();
	}
}

function addRandomValue() {
	const delta = Math.random()
		> 0.5 ? 1 : -1;
	data.push(curValue);
	curValue += delta;
}

File: demos/ruby/src/demo_random_walks.rb
data = []
curVal = 0

100.times{
  100.times{
    delta = rand > 0.5 ? 1 : -1
    data << curVal
    curVal += delta
    # visualize `data` here and press F5(or Continue button) over and over!
    binding.break
  }
}
